pilot/pkg/model/service.go

```diff
@@ -992,9 +992,20 @@ type ServiceInfo struct {
 	// PortNames provides a mapping of ServicePort -> port names. Note these are only used internally, not sent over XDS
 	PortNames map[int32]ServicePortName
 	// Source is the type that introduced this service.
-	Source kind.Kind
-	// Waypoint that clients should use when addressing traffic to this Service.
-	Waypoint string
+	Source   kind.Kind
+	Waypoint WaypointBindingStatus
+}
+
+type WaypointBindingStatus struct {
+	// ResourceName that clients should use when addressing traffic to this Service.
+	ResourceName string
+	// Error represents some error
+	Error *StatusMessage
+}
```

Comment:
Using it like a rust enum where if ResourceName then never Error? Should we document this (probably without using the word rust)?